script.R

```diff
@@ -0,0 +1,198 @@
+catnl <- function(x = "") cat(sprintf("%s\n", x))
+if_error <- function(x, y = NULL) {
+  res <- try(x, silent = TRUE)
+  if (is(res, "try-error")) {
+    return(y)
+  } else {
+    return(res)
+  }
+}
+`%||%` <- function(x, y) {
+  if (!length(x) || is.null(x)) y else x
+}
+
+args <- commandArgs(trailingOnly = TRUE)
+setwd(normalizePath(file.path(args[1])))
+number_of_workers <- as.integer(args[2])
+timeout <- as.integer(args[3])
+
+# Install required packages
+catnl("Installing required packages...")
+install.packages(c(
+  "pak"
+))
+pak::pkg_install(c(
+  "cli",
+  "miniCRAN",
+  "pkgbuild",
+  "pkgdepends",
+  "r-lib/revdepcheck",
+  "usethis",
+  "yaml"
+), ask = FALSE)
```

Comment:
we are GHA-caching both caches. We cannot use `crancache` only as we need to install a package from GH which is not supported there. I decided to use `pak` for installing all the required pkgs and we GHA-caching its cache so this should be super fast.
Cache mechanism from `pak` (which is a separate pkg `pkgache`) is much better than `crancache`. This package is very old and it seems that it has only `revdepcheck` is using this. Apparently, there is an issue in `revdepcheck` dated back at 2019 to replace `crancache` with `pkgcache` but it has not been done so far. Because of that - we have to put some extra effort to control this what goes where as well as store both of them. It's a little bit complicated unfortunately.